### PR TITLE
E2e gcs multifile

### PR DESCRIPTION
**Pull request by sumitsri12**
@sawantpritam Please review

### CODE REVIEW

File: src/e2e-test/resources/errorMessage.properties [hunk 1]
```diff
@@ -13,3 +13,11 @@ errorMessagePubSubMaxBatchSizeField=Invalid maximum batch size. Ensure the value
 errorMessagePubSubDelayThresholdField=Invalid delay threshold for publishing a batch. Ensure the value is a positive number.
 errorMessagePubSubRetryTimeout=Invalid max retry timeout for retrying failed publish. Ensure the value is a positive number.
 errorMessagePubSubErrorThreshold=Invalid error threshold for publishing. Ensure the value is a positive number.
+errorMessageTemporaryBucket=Bucket name can only contain lowercase letters, numbers, '.', '_', and '-'.
```
Comment (sawantpritam) [inline src/e2e-test/resources/errorMessage.properties]: Remove all unused error messages

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/tests/runner/package-info.java [hunk 2]
```diff
@@ -0,0 +1,4 @@
+/**
+ * Package contains the tests for the GCP features.
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/gcsmultifile/tests/runner/package-info.java]: Add proper comment

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/tests/runner/TestRunner.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
+/*
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/gcsmultifile/tests/runner/TestRunner.java]: Change package structure to ->
...plugin/gcsmultifile/runners/TestRunner.java

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/tests/runner/TestRunner.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+ * the License.
+ */
+package io.cdap.plugin.gcsmultifile.tests.runner;
+import io.cucumber.junit.Cucumber;
+import io.cucumber.junit.CucumberOptions;
+import org.junit.runner.RunWith;
+
+
+/**
+ * Test Runner to execute cases.
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/gcsmultifile/tests/runner/TestRunner.java]: Proper comment

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/tests/runner/TestRunner.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+ */
+@RunWith(Cucumber.class)
+@CucumberOptions(
+        features = {"src/e2e-test/features"},
+        glue = {"io.cdap.plugin.gcsmultifile.stepsdesign", "io.cdap.plugin.gcs.stepsdesign",
+          "io.cdap.plugin.bigquery.stepsdesign", "io.cdap.plugin.common.stepsdesign", "stepsdesign"},
+        tags = {"@MultiFile"},
+        monochrome = true,
+        plugin = {"pretty", "html:target/cucumber-html-report", "json:target/cucumber-reports/cucumber.json",
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/gcsmultifile/tests/runner/TestRunner.java]:
```
plugin = {"pretty", "html:target/cucumber-html-report/gcsmultifile", "json:target/cucumber-reports/cucumber-gcsmultifile.json",
          "junit:target/cucumber-reports/cucumber-gcsmultifile.xml"
```

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java [hunk 6]
```diff
@@ -0,0 +1,271 @@
...
+  public void enterGcsMultiFileSinkPropertyPathSuffix(String pathSuffix) {
+    CdfGcsActions.enterPathSuffix(PluginPropertyUtils.pluginProp(pathSuffix));
+  }
+
+  @Then("Toggle GcsMultiFile property allow flexible schemas to true")
+  public void toggleGcsMultiFilePropertyAllowFlexibleSchemasToTrue() throws IOException {
+    GCSMultiFileActions.selectAllowFlexibleSchema();
+  }
+
+
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java]: Remove extra line

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/locators/GCSMultiFileLocators.java [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+  @FindBy(how = How.XPATH, using = "//*[@data-cy='toggle-No']")
+  public static WebElement allowFlexible;
+
+  @FindBy(how = How.XPATH, using = "//*[@data-cy='select-format']")
+  public static WebElement format;
+
+  @FindBy(how = How.XPATH, using = "//*[@title='GCS Multi File']//following-sibling::div")
+  public static WebElement gcsMultiFileProperties;
+
+  @FindBy(how = How.XPATH, using = "//*[@class='fa fa-remove']")
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/gcsmultifile/locators/GCSMultiFileLocators.java]: `//a[@data-testid='close-config-popover']`

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/locators/GCSMultiFileLocators.java [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+  public static WebElement splitField;
+
+  @FindBy(how = How.XPATH, using = "//*[@data-cy='toggle-No']")
+  public static WebElement allowFlexible;
+
+  @FindBy(how = How.XPATH, using = "//*[@data-cy='select-format']")
+  public static WebElement format;
+
+  @FindBy(how = How.XPATH, using = "//*[@title='GCS Multi File']//following-sibling::div")
+  public static WebElement gcsMultiFileProperties;
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/gcsmultifile/locators/GCSMultiFileLocators.java]: Not required

File: src/e2e-test/features/gcsmultifile/GcsMultifile.feature [hunk 9]
```diff
@@ -0,0 +1,381 @@
...
+    Then Connect source as "GCS" and sink as "GCSMultiFiles" to establish connection
+    Then Open GCS source properties
+    Then Enter GCS property projectId and reference name
+    Then Enter GCS source property path "multiFileGCSBucket"
+    Then Toggle GCS source property skip header to true
+    Then Select GCS property format "csv"
+    Then Click on the Get Schema button
+    Then Validate "GCS" plugin properties
+    Then Close the GCS properties
+    Then Click on Source
```
Comment (sawantpritam) [inline src/e2e-test/features/gcsmultifile/GcsMultifile.feature]: Use from e2e framework -> `When Expand Plugin group in the LHS plugins list: "Source"`

File: src/e2e-test/features/gcsmultifile/GCSMultifileError.feature [hunk 10]
```diff
@@ -0,0 +1,35 @@
+@MultiFile
+Feature: GCS MultiFile Plugin Error scenarios
+
+  Scenario Outline: Verify mandatory fields error validation in GCS MultiFile sink plugin
+    Given Open Datafusion Project to configure pipeline
+    When Sink is GcsMultiFile
+    Then Open GcsMultiFile sink properties
+    Then Enter the GCSMultiFile properties with blank property "<property>"
+    Then Verify required property error for "<property>"
```
Comment (sawantpritam) [inline src/e2e-test/features/gcsmultifile/GCSMultifileError.feature]: Use from e2e framework -> `Then Validate mandatory property error for "<property>"`

File: src/e2e-test/features/gcsmultifile/BigQueryTOGCSMultifile_WithMacro.feature [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+    Then Enter BigQuery property "projectId" as macro argument "bqProjectId"
+    Then Enter BigQuery property "datasetProjectId" as macro argument "bqDatasetProjectId"
+    Then Enter BigQuery property "serviceAccountFilePath" as macro argument "bqServiceAccount"
+    Then Enter BigQuery property "dataset" as macro argument "bqDataset"
+    Then Enter BigQuery property "table" as macro argument "bqSourceTable"
+    Then Validate "BigQuery" plugin properties
+    Then Close the BigQuery properties
+    Then Open GcsMultiFile sink properties
+    Then Enter GcsMultiFile property reference name
+    Then Enter GCS MultiFile property "projectId" as macro argument "gcsMultiFileProjectId"
```
Comment (sawantpritam) [inline src/e2e-test/features/gcsmultifile/BigQueryTOGCSMultifile_WithMacro.feature]: (GCS MultiFile / GcsMultiFile)-> GCSMultiFile

Check all steps..

File: src/e2e-test/features/gcsmultifile/GcsMultifile.feature [hunk 12]
```diff
@@ -0,0 +1,381 @@
...
+    Then Open and capture pipeline preview logs
+    Then Verify the preview run status of pipeline in the logs is "succeeded"
+    Then Close the pipeline logs
+    Then Close the preview
+    Then Deploy the pipeline
+    Then Run the Pipeline in Runtime
+    Then Wait till pipeline is in running state
+    Then Open and capture logs
+    Then Verify the pipeline status is "Succeeded"
+    Then Verify data is transferred to target GCS bucket
```
Comment (sawantpritam) [inline src/e2e-test/features/gcsmultifile/GcsMultifile.feature]: `Then Verify data is transferred to target GCS bucket with file format "<formatType>"`

File: src/e2e-test/features/gcsmultifile/GcsMultifile.feature [hunk 13]
```diff
@@ -0,0 +1,381 @@
...
```
Comment (sawantpritam) [inline src/e2e-test/features/gcsmultifile/GcsMultifile.feature]: No NewLine at the end of file

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java [hunk 14]
```diff
@@ -86,8 +98,9 @@ public static void createBucketWithDataTypeTest2File() throws IOException, URISy
   }
 
   @After(order = 1, value = "@GCS_CSV_TEST or @GCS_TSV_TEST or @GCS_BLOB_TEST " +
-    "or @GCS_DELIMITED_TEST or @GCS_TEXT_TEST or @GCS_OUTPUT_FIELD_TEST or @GCS_DATATYPE_1_TEST or " +
-    "@GCS_DATATYPE_2_TEST")
+    "or @GCS_DELIMITED_TEST or @GCS_TEXT_TEST or @GCS_OUTPUT_FIELD_TEST or @GCS_CSV_MULTIPLE_SOURCE or " +
+    "@GCS_CSV_SPLIT_FIELD or @GCS_DATATYPE_1_TEST or " +
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java]: Remove @GCS_DATATYPE1 and 2 
Not part of this PR

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/actions/GCSMultiFileActions.java [hunk 15]
```diff
@@ -0,0 +1,87 @@
...
+import java.util.UUID;
+
+/**
+ * GCSMultiFile connector related actions.
+ */
+public class GCSMultiFileActions {
+
+  static {
+    SeleniumHelper.getPropertiesLocators(GCSMultiFileLocators.class);
+    SeleniumHelper.getPropertiesLocators(CdfStudioLocators.class);
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/gcsmultifile/actions/GCSMultiFileActions.java]: Remove not required `SeleniumHelper.getPropertiesLocators(CdfStudioLocators.class);`

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/locators/GCSMultiFileLocators.java [hunk 16]
```diff
@@ -0,0 +1,65 @@
...
+
+  @FindBy(how = How.XPATH, using = "//*[@class='fa fa-remove']")
+  public static WebElement closeButton;
+
+  @FindBy(how = How.XPATH, using = "//*[text()='Source ']")
+  public static WebElement source;
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/gcsmultifile/locators/GCSMultiFileLocators.java]: Not required

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java [hunk 17]
```diff
@@ -0,0 +1,271 @@
...
+  @Then("Enter the GCSMultiFile properties with blank property {string}")
+  public void enterTheGCSMultiFilePropertiesWithBlankProperty(String property) throws IOException {
+    if (property.equalsIgnoreCase("referenceName")) {
+      GCSMultiFileActions.enterGcsMultiFilepath(PluginPropertyUtils.pluginProp("multiFileGcsPath"));
+    } else if (property.equalsIgnoreCase("path")) {
+      GCSMultiFileActions.enterReferenceName();
+    } else if (property.equalsIgnoreCase("splitField")) {
+      GCSMultiFileActions.enterReferenceName();
+      GCSMultiFileActions.enterGcsMultiFilepath(PluginPropertyUtils.pluginProp("multiFileGcsPath"));
+      SeleniumHelper.replaceElementValue(GCSMultiFileLocators.splitField, "");
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java]: Use ElementHelper

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java [hunk 18]
```diff
@@ -0,0 +1,271 @@
...
+    } else {
+      Assert.fail("Invalid multiFile Mandatory Field " + property);
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java]: GCSMultiFile

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java [hunk 19]
```diff
@@ -0,0 +1,271 @@
...
+      .replace("BUCKET_NAME", PluginPropertyUtils.pluginProp(path));
+    String actualErrorMessage = PluginPropertyUtils.findPropertyErrorElement("path").getText();
+    Assert.assertEquals("Invalid bucket name error message should be displayed",
+                        expectedErrorMessage, actualErrorMessage);
+    String actualColor = PluginPropertyUtils.getErrorColor(PluginPropertyUtils.findPropertyErrorElement("path"));
+    String expectedColor = ConstantsUtil.ERROR_MSG_COLOR;
+    Assert.assertEquals(expectedColor, actualColor);
+  }
+
+  @Then("Enter the Gcs MultiFile Properties for table {string} and delimited format {string} and delimiter {string}")
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java]: Check entire file and remove unused step definitions

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java [hunk 20]
```diff
@@ -0,0 +1,271 @@
...
+  public void enterGcsMultiFilePropertyReferenceName() {
+    GCSMultiFileActions.enterReferenceName();
+  }
+
+  @Then("Select GcsMultiFile property format {string}")
+  public void selectGcsMultiFilePropertyFormat(String format) throws InterruptedException {
+    GCSMultiFileActions.selectFormat(format);
+  }
+
+  @Then("Enter the Gcs MultiFile Properties for path")
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java]: -> Enter GCSMultiFile property path

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java [hunk 21]
```diff
@@ -0,0 +1,271 @@
...
+  public void enterTheGcsMultiFilePropertiesForPath() throws IOException {
+    CdfGcsActions.getGcsBucket("gs://" + TestSetupHooks.gcsTargetBucketName);
+  }
+
+  @Then("Enter the Gcs MultiFile Properties for path {string}")
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java]: -> Enter GCSMultiFile property path {string}

File: src/e2e-test/features/gcsmultifile/BigQueryTOGCSMultifile_WithMacro.feature [hunk 22]
```diff
@@ -0,0 +1,61 @@
+@MultiFile
+Feature: GCS MultiFile sink - Verification of BigQuery to GCS MultiFile successful data transfer with macro arguments
+
+  @BQ_SOURCE_TEST @GCS_SINK_TEST
+  Scenario:Validate successful records transfer from BigQuery to GCS with macro arguments
```
Comment (sawantpritam) [inline src/e2e-test/features/gcsmultifile/BigQueryTOGCSMultifile_WithMacro.feature]: to GCSMultiFile

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java [hunk 23]
```diff
@@ -0,0 +1,271 @@
...
+  public void closeGcsMultiFileProperties() {
+    GCSMultiFileActions.closeGcsMultiFile();
+  }
+
+  @Then("Click on Source")
+  public void clickOnSource() {
+    GCSMultiFileActions.clickSource();
+  }
+
+  @Then("Verify Content Type Validation")
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java]: Verify GCSMultiFile Content Type validation error

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java [hunk 24]
```diff
@@ -0,0 +1,271 @@
...
+  }
+
+  @Then("Verify required property error for {string}")
+  public void verifyRequiredPropertyErrorFor(String property) {
+    CdfPluginPropertiesActions.clickValidateButton();
+    CdfPluginPropertiesActions.verifyRequiredPropertyHasNoValueErrorMessage(property);
+    CdfPluginPropertiesActions.verifyPluginPropertyInlineErrorMessageColor(property);
+  }
+
+  @Then("Verify invalid path name error message is displayed for path {string}")
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java]: -> Verify GCSMultiFile invalid path name........................

File: src/e2e-test/features/gcsmultifile/GcsMultifile.feature [hunk 25]
```diff
@@ -0,0 +1,381 @@
+@MultiFile
+Feature: MultiFile Sink - Verification of GCS MultiFile Sink plugin
```
Comment (sawantpritam) [inline src/e2e-test/features/gcsmultifile/GcsMultifile.feature]: GCSMultiFile Sink - Verification of successful data transfer using GCSMultiFile as sink

File: src/e2e-test/features/gcsmultifile/GCSMultifileError.feature [hunk 26]
```diff
@@ -0,0 +1,35 @@
+@MultiFile
+Feature: GCS MultiFile Plugin Error scenarios
```
Comment (sawantpritam) [inline src/e2e-test/features/gcsmultifile/GCSMultifileError.feature]: GCSMultiFile Sink - GCS MultiFile Plugin Error scenarios

**Review by sawantpritam — CHANGES_REQUESTED** [review]

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/tests/runner/package-info.java (hunk 2 above)
Comment (sumitsri12) [reply src/e2e-test/java/io/cdap/plugin/gcsmultifile/tests/runner/package-info.java]: Done

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/tests/runner/TestRunner.java (hunk 3 above)
Comment (sumitsri12) [reply src/e2e-test/java/io/cdap/plugin/gcsmultifile/tests/runner/TestRunner.java]: Done

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/tests/runner/TestRunner.java (hunk 4 above)
Comment (sumitsri12) [reply src/e2e-test/java/io/cdap/plugin/gcsmultifile/tests/runner/TestRunner.java]: Done

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/tests/runner/TestRunner.java (hunk 5 above)
Comment (sumitsri12) [reply src/e2e-test/java/io/cdap/plugin/gcsmultifile/tests/runner/TestRunner.java]: Done

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java (hunk 6 above)
Comment (sumitsri12) [reply src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java]: Done

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/locators/GCSMultiFileLocators.java (hunk 7 above)
Comment (sumitsri12) [reply src/e2e-test/java/io/cdap/plugin/gcsmultifile/locators/GCSMultiFileLocators.java]: Done

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/locators/GCSMultiFileLocators.java (hunk 8 above)
Comment (sumitsri12) [reply src/e2e-test/java/io/cdap/plugin/gcsmultifile/locators/GCSMultiFileLocators.java]: Removed

File: src/e2e-test/features/gcsmultifile/GcsMultifile.feature (hunk 9 above)
Comment (sumitsri12) [reply src/e2e-test/features/gcsmultifile/GcsMultifile.feature]: changed

File: src/e2e-test/features/gcsmultifile/GCSMultifileError.feature (hunk 10 above)
Comment (sumitsri12) [reply src/e2e-test/features/gcsmultifile/GCSMultifileError.feature]: changed

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java (hunk 14 above)
Comment (sumitsri12) [reply src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java]: @GCS_DATATYPE1 I am using in multifile for datatypes

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/actions/GCSMultiFileActions.java (hunk 15 above)
Comment (sumitsri12) [reply src/e2e-test/java/io/cdap/plugin/gcsmultifile/actions/GCSMultiFileActions.java]: Removed

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/locators/GCSMultiFileLocators.java (hunk 16 above)
Comment (sumitsri12) [reply src/e2e-test/java/io/cdap/plugin/gcsmultifile/locators/GCSMultiFileLocators.java]: Removed

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java (hunk 17 above)
Comment (sumitsri12) [reply src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java]: changed

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java (hunk 18 above)
Comment (sumitsri12) [reply src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java]: Changed

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java (hunk 19 above)
Comment (sumitsri12) [reply src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java]: Removed

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java (hunk 20 above)
Comment (sumitsri12) [reply src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java]: Changed

File: src/e2e-test/features/gcsmultifile/BigQueryTOGCSMultifile_WithMacro.feature (hunk 22 above)
Comment (sumitsri12) [reply src/e2e-test/features/gcsmultifile/BigQueryTOGCSMultifile_WithMacro.feature]: changed

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java (hunk 21 above)
Comment (sumitsri12) [reply src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java]: Changed

File: src/e2e-test/features/gcsmultifile/BigQueryTOGCSMultifile_WithMacro.feature (hunk 11 above)
Comment (sumitsri12) [reply src/e2e-test/features/gcsmultifile/BigQueryTOGCSMultifile_WithMacro.feature]: Changed

File: src/e2e-test/features/gcsmultifile/GCSMultifileError.feature (hunk 26 above)
Comment (sumitsri12) [reply src/e2e-test/features/gcsmultifile/GCSMultifileError.feature]: Done

File: src/e2e-test/features/gcsmultifile/GcsMultifile.feature (hunk 25 above)
Comment (sumitsri12) [reply src/e2e-test/features/gcsmultifile/GcsMultifile.feature]: Done

File: src/e2e-test/features/gcsmultifile/GcsMultifile.feature (hunk 12 above)
Comment (sumitsri12) [reply src/e2e-test/features/gcsmultifile/GcsMultifile.feature]: Done

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java (hunk 23 above)
Comment (sumitsri12) [reply src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java]: Done

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java (hunk 24 above)
Comment (sumitsri12) [reply src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java]: Done

File: src/e2e-test/resources/errorMessage.properties (hunk 1 above)
Comment (sumitsri12) [reply src/e2e-test/resources/errorMessage.properties]: Done

File: src/e2e-test/features/gcsmultifile/GcsMultifile.feature (hunk 13 above)
Comment (sumitsri12) [reply src/e2e-test/features/gcsmultifile/GcsMultifile.feature]: done

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java [hunk 27]
```diff
@@ -0,0 +1,268 @@
...
+      .replace("BUCKET_NAME", PluginPropertyUtils.pluginProp(path));
+    String actualErrorMessage = PluginPropertyUtils.findPropertyErrorElement("path").getText();
+    Assert.assertEquals("Invalid bucket name error message should be displayed",
+                        expectedErrorMessage, actualErrorMessage);
+    String actualColor = PluginPropertyUtils.getErrorColor(PluginPropertyUtils.findPropertyErrorElement("path"));
+    String expectedColor = ConstantsUtil.ERROR_MSG_COLOR;
+    Assert.assertEquals(expectedColor, actualColor);
+  }
+
+  @Then("Get the count of the records transferred")
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java]: Remove unused

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java [hunk 28]
```diff
@@ -0,0 +1,268 @@
...
+    enterPropertyAsMacroArgument(pluginProperty, macroArgument);
+  }
+
+  @Then("Enter runtime argument value for GCSMultiFile property path key {string}")
+  public void enterRuntimeArgumentValueForGCSMultiFilePropertyPathKey(String runtimeArgumentKey) {
+    ElementHelper.sendKeys(CdfStudioLocators.runtimeArgsValue(runtimeArgumentKey),
+                           "gs://" + TestSetupHooks.gcsTargetBucketName);
+  }
+
+  @Then("Verify data is transferred to target GCS bucket with file format {string}")
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java]: Remove unused

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java [hunk 29]
```diff
@@ -0,0 +1,268 @@
...
+    } else if (property.equalsIgnoreCase("splitField")) {
+      GCSMultiFileActions.enterReferenceName();
+      GCSMultiFileActions.enterGcsMultiFilepath(PluginPropertyUtils.pluginProp("multiFileGcsPath"));
+      ElementHelper.replaceElementValue(GCSMultiFileLocators.splitField, "");
+    } else {
+      Assert.fail("Invalid GCSMultiFile Mandatory Field " + property);
+    }
+  }
+
+  @Then("Verify required property error for {string}")
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java]: Remove unused

File: src/e2e-test/features/gcsmultifile/GCSMultifileError.feature (hunk 26 above)
Comment (sawantpritam) [reply src/e2e-test/features/gcsmultifile/GCSMultifileError.feature]: -> Feature: GCSMultiFile Sink - GCS MultiFile Plugin Error scenarios

File: src/e2e-test/features/gcsmultifile/BigQueryTOGCSMultifile_WithMacro.feature [hunk 30]
```diff
@@ -0,0 +1,61 @@
+@MultiFile
+Feature: GCS MultiFile sink - Verification of BigQuery to GCS MultiFile successful data transfer with macro arguments
```
Comment (sawantpritam) [inline src/e2e-test/features/gcsmultifile/BigQueryTOGCSMultifile_WithMacro.feature]: ->  Feature: GCSMultiFile Sink - Verification of BigQuery to GCS MultiFile successful data transfer with macro arguments

**Review by sawantpritam — CHANGES_REQUESTED** [review]

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java (hunk 27 above)
Comment (sumitsri12) [reply src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java]: removed

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java (hunk 28 above)
Comment (sumitsri12) [reply src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultiFileConnector.java]: using this in file format validation

**Pull request closed without merging**